algorithms/src/msm/variable_base/mod.rs

```diff
@@ -62,7 +80,36 @@ impl VariableBaseMSM {
                 }
             }
         }
-        standard::msm_standard(bases, scalars)
+        standard::msm(bases, scalars, strategy)
+    }
+
+    #[cfg(test)]
+    fn msm_naive<G: AffineCurve>(bases: &[G], scalars: &[<G::ScalarField as PrimeField>::BigInteger]) -> G::Projective {
+        use snarkvm_fields::Zero;
+        use snarkvm_utilities::BitIteratorBE;
+
+        let mut acc = G::Projective::zero();
+        for (base, scalar) in bases.iter().zip(scalars.iter()) {
+            acc += base.mul_bits(BitIteratorBE::new(*scalar));
+        }
+        acc
+    }
+
+    #[cfg(test)]
+    fn msm_naive_parallel<G: AffineCurve>(
+        bases: &[G],
+        scalars: &[<G::ScalarField as PrimeField>::BigInteger],
+    ) -> G::Projective {
+        use rayon::prelude::*;
+        use snarkvm_fields::Zero;
+        use snarkvm_utilities::BitIteratorBE;
+
+        bases
+            .par_iter()
+            .zip_eq(scalars.par_iter())
+            .map(|(base, scalar)| base.mul_bits(BitIteratorBE::new(*scalar)))
+            .fold(|| G::Projective::zero(), |a, b| a + b)
```

Comment:
This will still be slow, since I believe it's doing affine addition (which requires inverses)